server/http/http_server.cpp

```
@@ -317,11 +320,62 @@ transcript_response(
 
 }  // namespace
 
+// Magpie owns one mutable streaming workspace, so synthesis is serialized by the
+// runtime.  This coordinator sits in front of it when preemption is requested:
+// a newer request makes every older request ineligible to start (or continue)
+// and waits for the active request to release the runtime.
+class TtsPreemptionCoordinator {
+   public:
+    uint64_t claim() {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)